### PR TITLE
Added Vertex AI LLM class

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

This PR adds an LLM interface for Vertex AI models. It also fixes an import issue with the Embeddings class for Vertex AI.

## Type of Change
- [X] New feature
- [X] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [x] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [X] Unit tests
- [X] E2E tests
- [X] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [X] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [X] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_graphrag/llm/vertexai.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    def __init__(
+        self,
+        model_name: str = "gemini-1.5-flash-001",
+        model_params: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ):
+        if GenerativeModel is None or ResponseValidationError is None:
+            raise ImportError(
+                "Could not import Vertex AI python client. "
```
Comment (willtai) [inline src/neo4j_graphrag/llm/vertexai.py:49]: should python here be capitalised?

**Review by willtai — COMMENTED** [review]
The code looks good to me. Is updating documentation part of this PR too?

File: src/neo4j_graphrag/llm/vertexai.py (hunk 1 above)
Comment (alexthomas93) [reply src/neo4j_graphrag/llm/vertexai.py:49]: Good spot I've updated it now

**Comment by alexthomas93** [comment]
> The code looks good to me. Is updating documentation part of this PR too?

Good point I've updated it now

**Comment by alexthomas93** [comment]
@CodiumAI-Agent /update_changelog

**Review by willtai — APPROVED** [review]
🪢

**Pull request merged**